### ISSUE TITLE
Verify accessibility of TagSet popover link

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4357]
Discovered during the [accessibility audit](https://github.ibm.com/CDAI-design/pal/issues/1396).

## What will this achieve?

![Image](https://github.com/carbon-design-system/ibm-products/assets/1717398/7857be00-0278-4ff7-8a61-becb7fd5cdb4)

> **Violation**
> Text contrast of 2.31 with its background is less than the WCAG AA minimum >requirements for text of size 14px and weight of 400.
> The contrast ratio of text with its background must meet WCAG AA requirements

Investigate and improve accessibility of the link in the popover. Assuming we’re just using the same defaults as Carbon, in which case we should open a discussion with the `@carbon/react`. But should verify that we aren’t doing anything unique/custom. 

## How will success be measured?

Meet WCAG AA standards and increase contrast.

## Additional information

- Designs
- Existing code
- etc
